src/wazuh_modules/syscollector/CMakeLists.txt

```diff
@@ -65,6 +67,46 @@ link_directories(${SRC_FOLDER}/external/sqlite/)
 link_directories(${SRC_FOLDER}/external/cJSON/)
 link_directories(${SRC_FOLDER}/external/procps/)
 link_directories(${SRC_FOLDER}/external/bzip2/)
+link_directories(${SRC_FOLDER}/external/flatbuffers/build/)
+
+# add_custom_command does not create a new target. You have to define targets explicitly
+# by add_executable, add_library or add_custom_target in order to make them visible to make
+# For this reason, add_custom_target adds a target with the given name that executes the given commands.
+# The target has no output file and is always considered out of date even if the commands try to create a
+# file with the name of the target.
+
+if (NOT ${TARGET} STREQUAL "")
+  if(${TARGET} STREQUAL "server")
+    list(APPEND Schemas
+              common_agentInfo
+              hotfix_synchronization
+              hwinfo_syncrhonization
+              networkAddress_syncrhonization
+              networkIface_syncrhonization
+              osinfo_syncrhonization
+              package_synchronization
+              ports_syncrhonization
+              processes_syncrhonization
+              syscollector_synchronization
+    )
+    message("Compile schemas")
+    message("${SRC_FOLDER}/external/flatbuffers/build/flatc")
+    add_custom_target(compile_schemas)
+    foreach(schema IN LISTS Schemas)
+      set(FILE "${FLATBUFFERS_FOLDER}/${schema}.fbs")
+      message("Analizing file ${FILE}")
+      add_custom_command(
+          TARGET compile_schemas
+          COMMAND ${SRC_FOLDER}/external/flatbuffers/build/flatc
```

Comment:
I took the approach to make the schema compilation as simple as possible, and probably then see if some flags are necessary to add them, but for example, the force-string option converts the null values to empty. But in the implementation that Matias mentions I don't think is a good idea because in wdb_syscollector.c we have handles for null values not for empty ones.

Reply:
Perfect, let's keep the compilation without any flags, we can add them later =)